packages/sitecore-jss/src/site/graphql-redirects-service.test.ts

```diff
@@ -84,5 +89,49 @@ describe('GraphQLRedirectsService', () => {
 
       return expect(nock.isDone()).to.be.true;
     });
+
+    it('should cache fetch response', async () => {
+      mockRedirectsRequest(siteName);
+      const service = new GraphQLRedirectsService(config);
+      const redirectsResponse = await service.fetchRedirects();
+
+      expect(redirectsResponse).to.deep.equal(redirectsQueryResult.site?.siteInfo?.redirects);
+
+      nock.cleanAll();
+
+      nock(endpoint)
+        .post('/')
+        .reply(200, {
+          data: {
+            site: {},
+          },
+        });
+
+      const cachedResponse = await service.fetchRedirects();
+
+      expect(cachedResponse).to.deep.equal(redirectsResponse);
+    });
+
+    it('should be possible to disable cache', async () => {
+      mockRedirectsRequest(siteName);
+      const service = new GraphQLRedirectsService({ ...config, cacheEnabled: false });
+      const redirectsResponse = await service.fetchRedirects();
+
+      expect(redirectsResponse).to.deep.equal(redirectsQueryResult.site?.siteInfo?.redirects);
+
+      nock.cleanAll();
+
+      nock(endpoint)
+        .post('/')
+        .reply(200, {
+          data: {
+            site: {},
+          },
+        });
+
+      const cachedResponse = await service.fetchRedirects();
+
+      expect(cachedResponse).to.not.deep.equal(redirectsResponse);
+    });
   });
```

Comment:
Sorry I didn't mention this initially - could you please add some unit tests for exclude routes functionality, to ensure both  the built in and the optional function perform as expected?